### DISCUSSION TITLE
Lost in face-swap...

### DISCUSSION THREAD

**Discussion by 007SuperMario** [lllyasviel/Fooocus#2110] [Q&A]
Hello all,
I've been watching "tutorials" on YouTube and various forums for days to understand how to get something decent from the face-swap function, but I don't get anything good, at most a vaguely similar face...
The most "noob-proof" guide I tried is this:
https://www.reddit.com/r/fooocus/comments/1932tqh/face_swap/
But still I got nothing.
Let's take a practical case: can you swap any real face with this image created by an AI?

![il_1140xN 5088698801_6780](https://github.com/lllyasviel/Fooocus/assets/146724414/d9f083e8-b6ce-401c-bdc9-376b8490766b)

Thank you

**Comment by david-peter-smith** [comment]
Some fine points before doing anything:
- Make sure the image for the face you are swapping in is cropped to just the head and high resolution (at least 512x212)
- Make sure the image you are generating is 512x512 to 1024x1024 - you may need to mess with the preset file to add the custom aspect ratios if you don't have dimensions you like. Mine aspect ratios in my preset looks like this:

```
	"available_aspect_ratios": [
		"256*256","512*512","800*600","600*800","1024*768","768*1024","1024*1024","2048*1440","1440*2048","2048*2048"
    ]
```

## Method A:
At the bottom of the image prompt tab click advanced and then you will see weights appear under the image. You will need to experiment with these but they are set very conservative by default. bump them another .1 or more to get a closer match. Also you can take the resulting image and image paint that one a few times in a row until you get a closer match.

## Method B:
If you don't care about the general photo matching in composition and want to skip the inpaint part you can go to image prompt, have one box be the face swap and the second box be the source image. Always check "Advanced" at the bottom and play with the weights. Again you need to take the resulting image and replace the second box image prompt after the first go around 2-4 times before it looks really close the to face you are swapping in. I recommend if you use this approach to start with faceswap at 0.95 and 0.95 to start and go for quality.

- Using a good model and lora's for this approach helps produce a good result check out Civitai to get the latest Juggernaut  [https://civitai.com/ for the latest juggernaught ](https://civitai.com/models/133005/juggernaut-xl)

**Comment by infinity0** [comment]
I think the problem here is that the AI associates the surrounding Mario costume too strongly with a moustached face, which hides most of the facial features. That is, the AI does not identify Mario's face as something that can be swapped with other faces.

Even if you force the swapping via Inpainting (instead of Upscale as I described in #2644) and turn the Weight way up to 1.6, it insists on retaining the moustache that hides most of the facial features. But, hopefully you can kind of make out that yes in fact it is Elon Musk behind the moustache - I used the same photo of him as in #2644.

![2024-03-27_01-32-20_2540](https://github.com/lllyasviel/Fooocus/assets/78398/97cbb467-fcfc-4350-b6b3-880395559c10)
